### DISCUSSION TITLE
How to customize the outline display level

### DISCUSSION THREAD

**Discussion by KISGP** [web-infra-dev/rspress#1460] [Q&A]
I have many titles in the page, but by default, only h2-h4 can be displayed.

What should I do if I want H1 to show in the outline?
```markdown
# h1
## h2
### h3
#### h4
##### h5
###### h6

# h1
## h2
### h3
#### h4
##### h5
###### h6
```
![image](https://github.com/user-attachments/assets/705ee6ef-7585-43e1-b37c-9a6e8bee3b5b)

**Comment by Timeless0911** [comment]
There is no way to achieve this now. We only collect h2 ~ h4 as TOC (Table of Contents) in [mdx-rs](https://github.com/web-infra-dev/mdx-rs/blob/b6987e49fa01911d690525e46308f7a82dfa4beb/crates/plugin_toc/src/lib.rs#L3-L6).

I think we can add an option for TOC collection and add a config in Rspress then to set heading depth when collecting TOC in the future.

**Comment by maccuaa** [comment]
This would be great. In my use-case, I only want to show h2-h3 and not h4 in the outline
